rfcs/button-component.md

```diff
@@ -0,0 +1,81 @@
+# Button Component
+
+Rough DOM composition:
+
+```
++-------------------------------------------------------------------+
+| BaseButton (display: flex; justifyContent: space-between)         |
+| +-----------------+-----------------------+---------------------+ |
+| |  startEnhancer  |         label         |     endEnhancer     | |
+| |  display:flex   |                       |     display:flex    | |
+| |                 |                       |                     | |
+| |                 |                       |                     | |
+| |                 |                       |                     | |
+| +-----------------+-----------------------+---------------------+ |
++-------------------------------------------------------------------+
+```
+
+### Exports
+
+* `Button`
+* `StyledButton`
+* `StyledButtonLabel`
+* `StyledStartEnhancer`
+* `StyledEndEnhancer`
+* `KIND`
+* `SIZE`
+* `SHAPE`
+
+### <Button/> API
+
+* `label: node | function` - Optional
+  A label in the button.
+* `startEnhancer: node | function` - Optional
+  A helper rendered at the start of the button.
+* `endEnhancer: node | function` - Optional
+  A helper rendered at the end of the button.
+* `disabled: boolean` - Optional
+  Defines if the button disabled.
```

Comment:
Should we list `kind`, `size`, `shape` as well as their defaults here?

We also usually list the `overrides` prop here (see other component RFCs.

Reply:
Added!